### DISCUSSION TITLE
Review of the new filtering feature

### DISCUSSION THREAD

**Discussion by tissole** [aahnik/tgcf#131] [General]
With the new filter by file type feature implemented, I decided to run some tests to see how behaves. The current filters are: audio, gif, video, video_note, sticker, contact, photo, document, nofile. I was curious how the filters treat files like archives, books, or non-streamable files.

Filter video forwards only streamable videos. Videos uploaded as files or types unsupported by TG like mkv, avi, flv are not forwarded by this filter type.
Filter audio forwards the most common audio formats: mp3, flac, opus, aac, m4a, but not m4b.
Filter gif forwards only animated gifs, but not static gifs.

Filter sticker forwards stickers.
I have not tested filters video_note and contact(I don't even know what the last is).
Nofile forwards text messages, emojis, and links to TG posts.

Filter photo forwards only photos, but not photos send as files. This filter also forwards links to websites, Youtube, Mega, Udemy, and TG channels. Really!

So what about the rest of the formats? Non-streamable videos, videos sent as files, audiobooks in m4b format, photos sent as files, archives(zip, rar), books(pdf, epub, mobi), docs(doc, docx, rtf, txt)? All these are sent by the filter document.

So be careful not to blacklist filter document if you want to forward all the above or some of them. The document is the broadest category and further improvements could be made by refining the filtering here. Currently, if someone only wants a category will receive many more besides the desired one.

The script works as expected, the project has grown tremendously in the last months and now is a more mature, powerful, and versatile program. God work @aaknik!

Later I will post a review of the new delay feature. Many surprises :)

**Comment by mujahidazamcsm** [comment]
Hello..  

Is it possible to apply the different filtering for same source to different destinations ?
for e.g 
If we have below setup
S1 - D1, D2
S1 - D3 

Can we apply different filtering / replace for different forward sets?

so filter1/replace1 would apply for S1 - D1, D2 and filter2/replace2 would apply for S1 - D3 ?

Also was trying out the past feature and seems like the delay option was not considered. I had set it to 3 secs but even then messages forwarded without any delay and FloodWaitError was triggered.

**Reply by aahnik** [reply]
hi @mujahidazamcsm 

this is possible, but you need to run different instances of `tgcf`.

if you are running on computer, then make two separate folders with different config files, and start two separate terminals from them.

if you are running in a cloud platform, like heroku, you need to deploy two separate apps.

in one configuration file, all the sources must be unique.

-----

I will cover your scenario while making a video tutorial.
